File: GroupChat/app/src/main/java/com/example/groupchat/MainActivity.java
package com.example.groupchat;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private Button loginButton, registerButton, groupListButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 初始化按钮
        loginButton = findViewById(R.id.loginButton);
        registerButton = findViewById(R.id.registerButton);
        //groupListButton = findViewById(R.id.groupListButton);

        // 跳转到登录页面
        loginButton.setOnClickListener(view -> {
            Intent intent = new Intent(MainActivity.this, LoginActivity.class);
            startActivity(intent);
        });

        // 跳转到注册页面
        registerButton.setOnClickListener(view -> {
            Intent intent = new Intent(MainActivity.this, RegisterActivity.class);
            startActivity(intent);
        });

        // 跳转到群组列表页面
        //groupListButton.setOnClickListener(view -> {
        //    Intent intent = new Intent(MainActivity.this, GroupListActivity.class);
        //    startActivity(intent);
        //});
    }
}

File: GeoChatRoom/app/src/main/java/com/example/locationchat/activities/MainActivity.kt
package com.example.locationchat.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.locationchat.R
import com.example.locationchat.adapters.RoomListAdapter
import com.example.locationchat.databinding.ActivityMainBinding
import com.example.locationchat.services.LocationService

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var roomsAdapter: RoomListAdapter
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        
        // 初始化位置服务
        initLocationService()
        
        // 初始化聊天室列表
        setupRoomsList()
        
        // 监听附近的聊天室
        observeNearbyRooms()
    }
    
    private fun initLocationService() {
        // 检查并请求位置权限
        if (checkLocationPermission()) {
            startService(Intent(this, LocationService::class.java))
        } else {
            requestLocationPermission()
        }
    }
    
    private fun observeNearbyRooms() {
        // 使用 Firebase 或其他后端服务监听附近的聊天室
    }
}